From: Or Gerlitz <ogerlitz@mellanox.com>
Date: Tue, 18 Mar 2014 10:36:45 +0200
Subject: [PATCH v3 240/474] net/i40e: Avoid double setting of NETIF_F_SG for
 the HW encapsulation feature mask
Patch-mainline: v3.15-rc1
Git-commit: d70e941bff5f223017ba7001b8eb0423a636c070
References: bsc#909484 FATE#317397

The networking core does it for the driver during registration time.

Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/intel/i40e/i40e_main.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -6573,9 +6573,8 @@ static int i40e_config_netdev(struct i40
 	np = netdev_priv(netdev);
 	np->vsi = vsi;
 
-	netdev->hw_enc_features = NETIF_F_IP_CSUM	 |
-				  NETIF_F_TSO		 |
-				  NETIF_F_SG;
+	netdev->hw_enc_features |= NETIF_F_IP_CSUM	 |
+				  NETIF_F_TSO;
 
 	netdev->features = NETIF_F_SG		       |
 			   NETIF_F_IP_CSUM	       |
